scripts/quests/otherAreas/Rycharde_the_Chef.lua

```diff
@@ -0,0 +1,166 @@
+                end,
+            },
+        },
+    },
+
+    -- Section: Talk with Rycharde. Get quest (or decline). Handle "Take" optional dialog.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and vars.Prog == 2
+        end,
+
+        [xi.zone.MHAURA] =
+        {
+            ['Rycharde'] = quest:progressEvent(70, xi.items.DHALMEL_MEAT),
+            ['Take'] = quest:event(68), -- Optional dialog.
+
+            onEventFinish =
+            {
+                [70] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 3)
+                    if option == 71 or option == 72 then -- Accept quest.
+                        quest:begin(player)
+                    elseif option == 70 then -- Decline quest.
+                        quest:setVar(player, 'Prog', 3)      
+                    end
+                end,
+            },
+        },
+    },
+
+    -- Section: (OPTIONAL) Talk with Rycharde if you declined. Get quest (or decline again).
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and vars.Prog == 3
+        end,
+
+        [xi.zone.MHAURA] =
+        {
+            ['Rycharde'] = quest:progressEvent(71, xi.items.DHALMEL_MEAT), -- You rejected the quest and you ask again.
+
+            onEventFinish =
+            {
+                [71] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 3)
+                    if option == 71 or option == 72 then  -- Accept quest.
+                        quest:begin(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    -- Section: Quest accepted. Handle trade outcomes. Handle talk with Rycharde.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.MHAURA] =
+        {
+            ['Rycharde'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 3 then
+                        return quest:event(72) -- Interaction dialog.
+                    end
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, {{xi.items.DHALMEL_MEAT, 2}}) then
+                        return quest:progressEvent(74) -- Quest completed dialog.
+                    elseif npcUtil.tradeHasExactly(trade, {{xi.items.DHALMEL_MEAT, 1}}) then
+                        return quest:event(73) -- "That's not enough!" dialog.
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [74] = function(player, csid, option, npc)
+                    quest:complete(player)
+                    player:setCharVar("RychardeTheChefCompDay", VanadielDayOfTheYear()) -- Used for next quest wait time.
```

Comment:
Also out of scope of this pr but on eden we use a function to return vanaDay() a non repeating number that only increases per day so that there is no need to set 2 vars for day and year

Reply:
a function that does something similar to vanaYear * 1000 + vanaDayoftheyear. Would be useful to reduce vars.

Reply:
I wonder, how about something like (current time - epoch)/<length_of_vana_day>?

Reply:
function vanaDay()
    return VanadielYear() * 360 + VanadielDayOfTheYear()
end